### PR TITLE
Upgrade to Fedora 34

### LINKED ISSUE #69: Packer incompatible with Fedora 33 or Fedora 34
# 🐛 Summary #

[Fedora 33 updated the system-wide crypto policy to disallow SHA-1 hashes in signatures](https://fedoraproject.org/wiki/Changes/StrongCryptoSettings2).  This is a good thing, especially since [OpenSSH follows suit as of version 8.8](https://www.openssh.com/releasenotes.html).

Unfortunately, [Packer cannot support this change](https://github.com/hashicorp/packer/issues/8609) because of [an inflexibility in `go`](https://github.com/golang/go/issues/39885) (see also golang/go#36261 and golang/go#37278).  This is blocking us from upgrading our FreeIPA AMI to Fedora 33+, since Packer is unable to `ssh` to the instance launched from the base AMI after it is spun up.

It should be possible for us to workaround this limitation by [switching to a different system-wide crypto policy](https://fedoraproject.org/wiki/Changes/StrongCryptoSettings2#Upgrade.2Fcompatibility_impact) via the userdata fed to the base AMI when it is spun up.  This change can then be undone via an additional of a Packer provisioner in the `packer.json` configuration file.

### PR DESCRIPTION
**Pull request by jsf9k**
## 🗣 Description ##

This pull request upgrades the Packer build to use a Fedora 34 base AMI.  Resolves #69.

## 💭 Motivation and context ##

The packer build currently uses a Fedora 32 base AMI, which is no longer a supported OS.

## 🧪 Testing ##

I used these changes to build a new AMI for our staging environment.  I then successfully deployed the staging AMI and upgraded the FreeIPA cluster.

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - _eschew scope creep!_
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated to reflect the changes in this PR.
- [x] All new and existing tests pass.
- [x] Build a new staging AMI.
- [x] Test new staging AMI in our staging COOL environment.

## ✅ Pre-merge checklist ##

* [x] Finalize version.

## ✅ Post-merge checklist ##

* [x] Build a new production AMI.

### CODE REVIEW

**Review by felddy — APPROVED** [review]
Iron child out of Vulcan’s forge!  🜞

**Review by dav3r — APPROVED** [review]
LGTM! 👍

File: src/packer.json [hunk 1]
```diff
@@ -49,12 +49,13 @@
...
         "Base_AMI_Name": "{{ .SourceAMIName }}",
         "GitHub_Release_URL": "{{user `github_release_url`}}",
-        "OS_Version": "Fedora 32",
+        "OS_Version": "Fedora 34",
         "Pre_Release": "{{user `github_is_prerelease`}}",
         "Release": "{{user `github_release_tag`}}",
         "Team": "VM Fusion - Development"
       },
       "type": "amazon-ebs",
+      "user_data_file": "src/user_data.txt",
```
Comment (mcdonnnj) [inline src/packer.json]: Is there a reason this data is a `.txt`? I mean it makes no practical difference, I just like using extensions that give a clear idea on contents when possible.

**Review by mcdonnnj — APPROVED** [review]
Makes sense. Thanks for figuring this out 💪 I had one question just for clarity, but it's not a blocker.

File: src/packer.json (hunk 1 above)
Comment (jsf9k) [reply src/packer.json]: Please see commits a95cec8 and 16828a2.

Comment (mcdonnnj) [reply src/packer.json]: Excellent! Thank you.

**Pull request merged**